crates/re_space_view_spatial/src/visualizers/points3d.rs

```diff
@@ -187,72 +194,81 @@ impl VisualizerSystem for Points3DVisualizer {
     fn execute(
         &mut self,
         ctx: &ViewerContext<'_>,
-        query: &ViewQuery<'_>,
+        view_query: &ViewQuery<'_>,
         view_ctx: &ViewContextCollection,
     ) -> Result<Vec<re_renderer::QueueableDrawData>, SpaceViewSystemExecutionError> {
-        let num_points = super::entity_iterator::count_instances_in_archetype_views::<
-            Points3DVisualizer,
-            Points3D,
-            8,
-        >(ctx, query);
-
-        if num_points == 0 {
-            return Ok(Vec::new());
-        }
-
         let mut point_builder = PointCloudBuilder::new(ctx.render_ctx);
-        point_builder.reserve(num_points)?;
         point_builder
             .radius_boost_in_ui_points_for_outlines(SIZE_BOOST_IN_POINTS_FOR_POINT_OUTLINES);
 
-        // We need lines from keypoints. The number of lines we'll have is harder to predict, so we'll go with the dynamic allocation approach.
+        // We need lines from keypoints. The number of lines we'll have is harder to predict, so we'll go
+        // with the dynamic allocation approach.
         let mut line_builder = LineDrawableBuilder::new(ctx.render_ctx);
         line_builder
             .radius_boost_in_ui_points_for_outlines(SIZE_BOOST_IN_POINTS_FOR_POINT_OUTLINES);
 
-        super::entity_iterator::process_archetype_pov1_comp5::<
-            Points3DVisualizer,
-            Points3D,
-            Position3D,
-            Color,
-            Radius,
-            Text,
-            re_types::components::KeypointId,
-            re_types::components::ClassId,
-            _,
-        >(
+        super::entity_iterator::process_archetype::<Points3DVisualizer, Points3D, _>(
             ctx,
-            query,
+            view_query,
             view_ctx,
             view_ctx.get::<EntityDepthOffsets>()?.points,
-            |_ctx,
-             ent_path,
-             _ent_props,
-             ent_context,
-             (_time, _row_id),
-             instance_keys,
-             positions,
-             colors,
-             radii,
-             labels,
-             keypoint_ids,
-             class_ids| {
-                let data = Points3DComponentData {
-                    instance_keys,
-                    positions,
-                    colors,
-                    radii,
-                    labels,
-                    keypoint_ids,
-                    class_ids,
+            |ctx, entity_path, _entity_props, spatial_ctx, results| {
+                re_tracing::profile_scope!(format!("{entity_path}"));
+
+                use crate::visualizers::CachedRangeResultsExt as _;
+
+                let resolver = ctx.recording().resolver();
+
+                let positions = match results.get_dense::<Position3D>(resolver) {
+                    Some(Ok(positions)) => positions,
+                    Some(err @ Err(_)) => err?,
```

Comment:
nit: a bit shorter/simpler here and elsewhere

```suggestion
                    Some(positions) => positions?,
```